.ci_support/build_all.py

```diff
@@ -68,6 +73,34 @@ def build_all(recipes_dir, arch):
                 if not built:
                     old_comp_folders.append(folder)
                     continue
+            if platform == 'osx' and (
+                    'MACOSX_DEPLOYMENT_TARGET' in text or
+                    'MACOSX_SDK_VERSION' in text):
+                deployment_version = (0, 0)
+                sdk_version = (0, 0)
+                config = safe_load(text)
+
+                with open(variant_config_file, 'r') as f:
+                    variant_text = ''.join(f.readlines())
+                variant_config = safe_load(variant_text)
+
+                if 'MACOSX_DEPLOYMENT_TARGET' in config:
+                    for version in config['MACOSX_DEPLOYMENT_TARGET']:
+                        version = tuple([int(x) for x in version.split('.')])
+                        deployment_version = max(deployment_version, version)
+                if 'MACOSX_SDK_VERSION' in config:
+                    for version in config['MACOSX_SDK_VERSION']:
+                        version = tuple([int(x) for x in version.split('.')])
+                        sdk_version = max(deployment_version, version)
+
+                max_version = max(deployment_version, sdk_version)
+                if max_version != (0, 0):
+                    max_version = '.'.join([str(x) for x in max_version])
+                    variant_config['MACOSX_DEPLOYMENT_TARGET'] = [max_version]
+                    variant_config['MACOSX_SDK_VERSION'] = [max_version]
+                    with open(variant_config_file, 'w') as f:
+                        new_file = safe_dump(variant_config)
+                        f.write(new_file)
```

Comment:
I think the maximum then needs to be added at the end of the loop instead, i.e. move this

```suggestion
```

and add this before `if old_comp_folders:`

```python
    with open(variant_config_file, 'r') as f:
        variant_text = ''.join(f.readlines())
    variant_config = safe_load(variant_text)
    if max_version != (0, 0):
        print("Overriding MACOSX_DEPLOYMENT_TARGET to be", deployment_version)
        variant_config['MACOSX_DEPLOYMENT_TARGET'] = ['.'.join(map(str, deployment_version))]
    if max_version != (0, 0):
        print("Overriding MACOSX_SDK_VERSION to be", sdk_version)
        variant_config['MACOSX_SDK_VERSION'] = ['.'.join(map(str, sdk_version))]
    with open(variant_config_file, 'w') as f:
        f.write(safe_dump(variant_config))
```